pkg/controller/types/group.go

```diff
@@ -49,36 +49,43 @@ type GroupSelector struct {
 	// If Namespace and NamespaceSelector both are unset, it selects the ExternalEntities in all the Namespaces.
 	// TODO: Add validation in API to not allow externalEntitySelector and podSelector in the same group.
 	ExternalEntitySelector labels.Selector
+
+	NodeSelector labels.Selector
 }
 
-func NewGroupSelector(namespace string, podSelector, nsSelector, extEntitySelector *metav1.LabelSelector) *GroupSelector {
+// NewGroupSelector converts the podSelector, namespaceSelector and externalEntitySelector
+// and NetworkPolicy Namespace to a networkpolicy.GroupSelector object.
+func NewGroupSelector(namespace string, podSelector, nsSelector, extEntitySelector *metav1.LabelSelector, nodeSelectors ...*metav1.LabelSelector) *GroupSelector {
 	groupSelector := GroupSelector{}
 	if podSelector != nil {
-		pSelector, _ := metav1.LabelSelectorAsSelector(podSelector)
-		groupSelector.PodSelector = pSelector
+		groupSelector.PodSelector, _ = metav1.LabelSelectorAsSelector(podSelector)
 	}
 	if extEntitySelector != nil {
-		eSelector, _ := metav1.LabelSelectorAsSelector(extEntitySelector)
-		groupSelector.ExternalEntitySelector = eSelector
+		groupSelector.ExternalEntitySelector, _ = metav1.LabelSelectorAsSelector(extEntitySelector)
 	}
 	if nsSelector == nil {
 		// No namespaceSelector indicates that the pods must be selected within
 		// the NetworkPolicy's Namespace.
 		groupSelector.Namespace = namespace
 	} else {
-		nSelector, _ := metav1.LabelSelectorAsSelector(nsSelector)
-		groupSelector.NamespaceSelector = nSelector
+		groupSelector.NamespaceSelector, _ = metav1.LabelSelectorAsSelector(nsSelector)
+	}
+
+	if len(nodeSelectors) > 0 && nodeSelectors[0] != nil {
```

Comment:
this is a proof why variadic function doesn't apply here.

Reply:
Updated.